### PR TITLE
Jest migration

### PR DESCRIPTION
**Pull request by SeanArmstrong**

### CODE REVIEW

**Comment by toni-leigh** [comment]
squash all the things

File: src/components/dialog/__spec__.js [hunk 1]
```diff
@@ -209,17 +219,17 @@ describe('Dialog', () => {
 
     describe('when animating', () => {
       beforeEach(() => {
-        jasmine.clock().install();
+        jest.useFakeTimers();
       });
 
       afterEach(() => {
-        jasmine.clock().uninstall();
+        jest.clearAllTimers();
```
Comment (andrewjtait) [inline src/components/dialog/__spec__.js]: instead of clearing the timers, shouldn't this set to `useRealTimers`?

**Comment by SeanArmstrong** [comment]
The following implementation files have been changed as part of this work

* form.js
* portrait.js
* rainbow.js
* table-ajax.js
* tabs.js
* input-validations.js
* input.js
* flux.js
* poller.js

**Comment by SeanArmstrong** [comment]
Cross browser tested components above the Carbon Demo Site

**Pull request merged**